lib/msf/core/exploit/remote/ldap/server.rb

```diff
@@ -0,0 +1,115 @@
+# -*- coding: binary -*-
+
+module Msf
+
+###
+#
+# This module exposes methods for querying a remote LDAP service
+#
+###
+module Exploit::Remote::LDAP
+module Server
+  include Exploit::Remote::SocketServer
+
+  #
+  # Initializes an exploit module that serves LDAP requests
+  #
+  def initialize(info = {})
+    super
+
+    register_options(
+      [
+        OptPort.new('SRVPORT', [true, 'The local port to listen on.', 389]),
+        OptPath.new('LDIF_FILE', [ false, "Directory LDIF file path"]),
+      ], Exploit::Remote::LDAP::Server
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('LdapServerUdp', [true, "Serve UDP LDAP requests", true]),
+        OptBool.new('LdapServerTcp', [true, "Serve TCP LDAP requests", true])
+      ], Exploit::Remote::LDAP::Server
+    )
+  end
+
+  attr_accessor :service # :nodoc:
+
+  #
+  # Read LDIF file - from net-ldap ldapserver.rb#L162
+  #
+  # @ return [Hash] parsed ldif file
+  def read_ldif
+    if !datastore['LDIF_FILE'].blank? && File.exists?(datastore['LDIF_FILE'])
+      ary = File.readlines(datastore['LDIF_FILE'])
+      ldif = {}
+      while (line = ary.shift) && line.chomp!
+        if line =~ /^dn:[\s]*/i
+          dn = $'
+          ldif[dn] = {}
+          while (attrib = ary.shift) && attrib.chomp! && attrib =~ /^([\w]+)[\s]*:[\s]*/
+            ldif[dn][$1] ||= []
+            ldif[dn][$1] << $'
+          end
+        end
+      end
+      return ldif
+    else
+      nil
+    end
```

Comment:
With the guard clause above

```suggestion
```

Reply:
Sorry not quite grasping pre-coffee - whats the concern here? Getting :coffee:

Reply:
Got this one when i saw the code together - disregard, resolved